catalog/app/containers/Bucket/Listing.js

```diff
@@ -264,16 +340,129 @@ const useListingStyles = M.makeStyles((t) => ({
+            <M.CircularProgress />
+          </div>
+        )}
+        {!items.length && !prefix ? (
+          <M.Typography className={classes.empty} variant="h5">
+            No files
+          </M.Typography>
+        ) : (
+          <>
+            <HeaderWithPrefixFiltering
+              {...{
+                items,
+                prefixValue,
+                changePrefixValue,
+                commitPrefixChange,
+                truncated,
+              }}
+            />
+            <div ref={scrollRef} />
+            {!items.length && (
+              <div className={classes.noMatch}>
+                No entries starting with{' '}
+                <b>
+                  <code>{`"${prefix}"`}</code>
+                </b>
+              </div>
+            )}
+            {pagination.paginated.map(
+              ListingItem.case({
+                Dir: ({ name, to }) => (
+                  <Item icon="folder_open" key={name} name={name || EMPTY} to={to} />
+                ),
+                File: ({ name, to, size, modified, archived }) => (
+                  <Item
+                    icon="insert_drive_file"
+                    key={name}
+                    name={name}
+                    to={to}
+                    archived={archived}
+                  >
+                    <div className={classes.size}>{readableBytes(size)}</div>
+                    {!!modified && (
+                      <div className={classes.modified}>{modified.toLocaleString()}</div>
+                    )}
+                  </Item>
+                ),
+              }),
+            )}
+            {(truncated || pagination.pages > 1) && (
+              <M.Box>
+                <M.Divider />
+                <M.Box display="flex" alignItems="center" minHeight={36} px={1}>
+                  {truncated && (
+                    <M.Typography variant="body2" component="span" color="textSecondary">
+                      <M.Icon
+                        style={{
+                          fontSize: 16,
+                          verticalAlign: 'text-bottom',
+                          marginRight: 4,
+                        }}
+                      >
+                        warning
+                      </M.Icon>
+                      Results truncated to {totalItems} items
+                    </M.Typography>
+                  )}
+                  <M.Box flexGrow={1} />
+                  {pagination.pages > 1 && <Pagination.Controls {...pagination} />}
+                </M.Box>
+              </M.Box>
+            )}
+          </>
+        )}
+      </M.CardContent>
+    </M.Card>
+  )
+}
+
+export function ListingWithLocalFiltering({ items, truncated = false, locked = false }) {
+  const classes = useListingStyles()
+
+  const scrollRef = React.useRef(null)
+  const scroll = React.useCallback(
+    (prev) => {
+      if (prev && scrollRef.current) scrollRef.current.scrollIntoView()
+    },
+    [scrollRef],
+  )
```

Comment:
I think it is worth to separate this into custom hook (e.g. `const [scroll, scrollRef] = useScrollIntoView()`)

Reply:
yeah, seems like a good idea